cluster-autoscaler/core/scaledown/eligibility/eligibility.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eligibility
+
+import (
+	"reflect"
+	"time"
+
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/core/scaledown/actuation"
+	"k8s.io/autoscaler/cluster-autoscaler/core/scaledown/unremovable"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/utilization"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/gpu"
+
+	apiv1 "k8s.io/api/core/v1"
+	klog "k8s.io/klog/v2"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+const (
+	// ScaleDownDisabledKey is the name of annotation marking node as not eligible for scale down.
+	ScaleDownDisabledKey = "cluster-autoscaler.kubernetes.io/scale-down-disabled"
+)
+
+// Checker is responsible for deciding which nodes pass the criteria for scale down.
+type Checker struct {
+	context          *context.AutoscalingContext
+	thresholdGetter  utilizationThresholdProcessor
+	unremovableNodes *unremovable.Nodes
+}
+
+type utilizationThresholdProcessor interface {
```

Comment:
Elsewhere in the code we use processors directly (and generally take them from context rather than pass individual processors to functions I think). I'm fine with the pattern of using local interface in general, but in this case it seems inconsistent and maybe an overkill (especially because you don't have any unittests that would use this interface for cleaner mocking, which I think is a major reason for this pattern).

I don't feel super strongly about the interface argument, but if you want to keep it please consider renaming it. I think xxxYyyProcessor suggests to the reader it is a processor interface that they could implement, rather than just a local interface for parts of a different processor.